releases/latest_version.go

```diff
@@ -156,13 +161,18 @@ func (lv *LatestVersion) Remove(ctx context.Context) error {
 }
 
 func (lv *LatestVersion) findLatestMatchingVersion(pvs rjson.ProductVersionsMap, vc version.Constraints) (*rjson.ProductVersion, bool) {
+	requiredMetadata := lv.Enterprise.requiredMetadata()
 	versions := make(version.Collection, 0)
 	for _, pv := range pvs.AsSlice() {
 		if !lv.IncludePrereleases && pv.Version.Prerelease() != "" {
 			// skip prereleases if desired
 			continue
 		}
 
+		if pv.Version.Metadata() != requiredMetadata {
```

Comment:
```suggestion
	versions := make(version.Collection, 0)
	for _, pv := range pvs.AsSlice() {
		if !lv.IncludePrereleases && pv.Version.Prerelease() != "" {
			// skip prereleases if desired
			continue
		}

		if pv.Version.Metadata() != lv.Enterprise.requiredMetadata() {
```
Nitpick - we can simplify this a little bit.

Reply:
I had placed `requiredMetadata := lv.Enterprise.requiredMetadata()` outside of the loop since the returned value never changes - repeatedly calling it within the loop seemed like unnecessary overhead.  And now that `lv.Enterprise` is a pointer (that could be `nil`) we can't avoid setting a variable:

```go
requiredMetadata := ""
if lv.Enterprise != nil {
	requiredMetadata = lv.Enterprise.requiredMetadata()
}
```

So if we need that variable now, I'd propose keeping it outside of the loop to make the loop contents easier to reason through (and avoid unnecessary work).  WDYT?